src/wp-includes/meta.php

```diff
@@ -1414,6 +1417,7 @@ function register_meta( $object_type, $meta_key, $args, $deprecated = null ) {
 		'sanitize_callback' => null,
 		'auth_callback'     => null,
 		'show_in_rest'      => false,
+		'revisions_enabled' => false,
```

Comment:
We could add a `_doing_it_wrong` if you pass `revisions_enabled` and type that was not post. We could even add a message saying that post type does not support revisions.

Reply:
I think you mean for a post type that doesn't support revisions which any post type can opt into with supports=>revisions. makes sense.

Reply:
Added a check for this condition in [`f9b760b` (#4859)](https://github.com/WordPress/wordpress-develop/pull/4859/commits/f9b760b7d5fb9de3e56e4ebadb60c9fcda4c6ccd)

Added a test to validate the check works correctly in [`8360faa` (#4859)](https://github.com/WordPress/wordpress-develop/pull/4859/commits/8360faa26e9aa07718aba603d37f62f70039615f)

Reply:
Now emits `Meta keys cannot enable revisions support unless the object type supports revisions.`